pulp_smash/utils.py

```diff
@@ -86,6 +86,23 @@ def os_is_f27(cfg, pulp_host=None):
     return response.returncode == 0
 
 
+def os_is_fips(cfg, pulp_host=None):
+    """Return ``True`` If the server has fips enabled, or ``False`` otherwise.
+
+    :param pulp_smash.config.PulpSmashConfig cfg: Information about the system
+        being targeted.
+    :param pulp_host: A :class:`pulp_smash.config.PulpHost` to target,
+        instead of the default chosen by :class:`pulp_smash.cli.Client`.
+    :returns: True or false.
+    """
+    response = cli.Client(cfg, cli.echo_handler, pulp_host).run((
+        'grep',
+        '1',
+        '/proc/sys/crypto/fips_enabled'
```

Comment:
There's a bunch of subtle problems inherent in checking the value of `/proc/sys/crypto/fips_enabled` in this manner. In order of decreasing importance:

* What would happen if this file contained a value of "10?" This method would return true. However, we don't know the meaning of "10," so why would we go ahead and state that FIPS is enabled when we don't know whether this is true or not?
* Using grep in this manner makes it impossible to distinguish between several different error conditions. These include:

  * `/proc/sys/crypto/fips_enabled` doesn't exist.
  * `/proc/sys/crypto/fips_enabled` exists but is unreadable.
  * `/proc/sys/crypto/fips_enabled` exists and is readable, but doesn't contain "1."

* Some of the files in `/proc/sys` are deprecated. The `sysctl` executable prevents access to deprecated files in this hierarchy. Directly accessing these files bypasses these protections.